### PR TITLE
Getting Started improvements

### LINKED ISSUE #845: Document creating API key for webhooks
#### Summary

Problem when downlink message via webhooks:
**error:pkg/auth/rights:no_application_rights (no rights for application `app3`)**

...

#### Steps to Reproduce

1. ...Clone lorawan-stack from github (as 24.06.2019) + gettingstarted instruction
2. ... add app, dev, webhook, key as in gettingstarted
3. ... add all ACLs to application
4... webhook successfully send POST data
5.. Insufficient access rights to put data

root@propc:/home/robert# **ttn-lw-cli applications create app3 --user-id admin**

{
  "ids": {
    "application_id": "app3"
  },
  "created_at": "2019-06-24T12:21:16.862Z",
  "updated_at": "2019-06-24T12:21:16.862Z"
}

root@propc:/home/robert# **ttn-lw-cli end-devices create app3 dev4 --dev-eui 10217E00000003FF --app-eui A000000000000002 --frequency-plan-id EU_863_870 --root-keys.app-key.key 44D70CFA703C740D8EAB6A2811732B76 --lorawan-version 1.0.2 --lorawan-phy-version 1.0.2-b**

Flag --app-eui has been deprecated, use the join-eui flag

{
  "ids": {
    "device_id": "dev4",
    "application_ids": {
      "application_id": "app3"
    },
    "dev_eui": "10217E00000003FF",
    "join_eui": "A000000000000002"
  },
  "created_at": "2019-06-24T12:21:34.430Z",
  "updated_at": "2019-06-24T12:21:34.667620964Z",
  "attributes": {
  },
  "network_server_address": "localhost",
  "application_server_address": "localhost",
  "join_server_address": "localhost",
  "lorawan_version": "1.0.2",
  "lorawan_phy_version": "1.0.2-b",
  "frequency_plan_id": "EU_863_870",
  "supports_join": true,
  "root_keys": {
    "app_key": {
      "key": "44D70CFA703C740D8EAB6A2811732B76"
    }
  }
}

root@propc:/home/robert# **ttn-lw-cli applications api-keys create --name link --application-id app3 --right-application-link --right-application-all --right-application-delete --right-application-devices-read --right-application-devices-read-keys --right-application-devices-write --right-application-devices-write-keys --right-application-info --right-application-link --right-application-settings-api-keys --right-application-settings-basic --right-application-settings-collaborators --right-application-traffic-down-write --right-application-traffic-read --right-application-traffic-up-write**

  INFO API key ID: 4DU2SXFXICLJUGHJPAC6BUFYFHZSKT6SOBXABUQ
  INFO API key value: NNSXS.4DU2SXFXICLJUGHJPAC6BUFYFHZSKT6SOBXABUQ.KNZJFT6R3DYLI5N5XZJFWCXDDYJBF2HAXHOCK2TXDBMTFCET3KFQ
  WARN The API key value will never be shown again
  WARN Make sure to copy it to a safe place
{
  "id": "4DU2SXFXICLJUGHJPAC6BUFYFHZSKT6SOBXABUQ",
  "key": "NNSXS.4DU2SXFXICLJUGHJPAC6BUFYFHZSKT6SOBXABUQ.KNZJFT6R3DYLI5N5XZJFWCXDDYJBF2HAXHOCK2TXDBMTFCET3KFQ",
  "name": "link",
  "rights": [
    "RIGHT_APPLICATION_DEVICES_WRITE",
    "RIGHT_APPLICATION_SETTINGS_BASIC",
    "RIGHT_APPLICATION_SETTINGS_COLLABORATORS",
    "RIGHT_APPLICATION_TRAFFIC_UP_WRITE",
    "RIGHT_APPLICATION_DELETE",
    "RIGHT_APPLICATION_DEVICES_READ",
    "RIGHT_APPLICATION_LINK",
    "RIGHT_APPLICATION_TRAFFIC_READ",
    "RIGHT_APPLICATION_DEVICES_READ_KEYS",
    "RIGHT_APPLICATION_DEVICES_WRITE_KEYS",
    "RIGHT_APPLICATION_ALL",
    "RIGHT_APPLICATION_INFO",
    "RIGHT_APPLICATION_SETTINGS_API_KEYS",
    "RIGHT_APPLICATION_TRAFFIC_DOWN_WRITE"
  ]
}

 

root@propc:/home/robert# **ttn-lw-cli applications link set app3 --api-key** NNSXS.GIPVAVIAO77RFBOWTDO43XIZ5SDTRPDX7QKOGCQ.EJJRVXWEX7LIFSPYMT4WBPPSCIMIJEXTRMFOSRNT47RPF6KTWCJQ

{
  "api_key": "NNSXS.GIPVAVIAO77RFBOWTDO43XIZ5SDTRPDX7QKOGCQ.EJJRVXWEX7LIFSPYMT4WBPPSCIMIJEXTRMFOSRNT47RPF6KTWCJQ"
}

 
 

root@propc:/home/robert# **ttn-lw-cli applications webhooks set --application-id app3 --webhook-id wh4 --format json   --base-url http://192.168.0.8/IoT/   --join-accept.path up.php   --uplink-message.path up.php**

{
  "ids": {
    "application_ids": {
     "application_id": "app3"
    },
    "webhook_id": "wh4"
  },
  "created_at": "2019-06-24T12:34:24.986984352Z",
  "updated_at": "2019-06-24T12:34:37.766238582Z",
  "base_url": "http://192.168.0.8/IoT/",
  "format": "json",
  "uplink_message": {
    "path": "up.php"
  },
  "join_accept": {
    "path": "up.php"
  }
}

 

root@propc:/home/robert# **curl http://localhost:1885/api/v3/as/applications/app3/webhooks/wh4/devices/dev4/down/push -X POST   -H 'Authorization: Bearer "NNSXS.GIPVAVIAO77RFBOWTDO43XIZ5SDTRPDX7QKOGCQ.EJJRVXWEX7LIFSPYMT4WBPPSCIMIJEXTRMFOSRNT47RPF6KTWCJQ'  --data '{"downlinks":[{"frm_payload":"vu8=","f_port":15,"priority":"NORMAL"}]}'**

**_error:pkg/auth/rights:no_application_rights (no rights for application `app3`)_**

                                               --as.link-mode="all"
                                                --as.mqtt.listen=":1883"
                                            --as.mqtt.listen-tls=":8883"
                                        --as.webhooks.queue-size="16"
                                            --as.webhooks.target="direct"
                                           --as.webhooks.timeout="5s"
                                           --as.webhooks.workers="16"
                                        --blob.aws.access-key-id=""
                                             --blob.aws.endpoint=""
                                               --blob.aws.region=""
                                    --blob.aws.secret-access-key=""
                                        --blob.aws.session-token=""
                                          --blob.gcp.credentials=""
                                     --blob.gcp.credentials-file=""
                                          --blob.local.directory="./public/blob"
                                                 --blob.provider="local"
                                               --cluster.address=""
                                    --cluster.application-server=""
                                        --cluster.gateway-server=""
                                       --cluster.identity-server=""
                                                  --cluster.join=""
                                           --cluster.join-server=""
                                                  --cluster.keys="db114f80fd0ebf2a7b69db7e5a56fce248e53600e256b3a85d5b0ab844bc1aa8"
                                                  --cluster.name=""
                                        --cluster.network-server=""
                                                   --cluster.tls="false"
                                                        --config="/root/.ttn-lw-stack.yml"
                                                 --console.mount=""
                                   --console.oauth.authorize-url="http://localhost:1885/oauth/authorize"
                                       --console.oauth.client-id="console"
                                   --console.oauth.client-secret="console"
                                       --console.oauth.token-url="http://localhost:1885/oauth/token"
                                        --console.ui.as.base-url="http://localhost:1885/api/v3"
                                         --console.ui.as.enabled="true"
                                    --console.ui.assets-base-url="/assets"
                                      --console.ui.canonical-url="http://localhost:1885/console"
                                           --console.ui.css-file="console.css"
                                       --console.ui.descriptions=""
                                        --console.ui.gs.base-url="http://localhost:1885/api/v3"
                                         --console.ui.gs.enabled="true"
                                        --console.ui.icon-prefix="console-"
                                        --console.ui.is.base-url="http://localhost:1885/api/v3"
                                         --console.ui.is.enabled="true"
                                            --console.ui.js-file="console.js"
                                        --console.ui.js.base-url="http://localhost:1885/api/v3"
                                         --console.ui.js.enabled="true"
                                           --console.ui.language="en"
                                        --console.ui.ns.base-url="http://localhost:1885/api/v3"
                                         --console.ui.ns.enabled="true"
                                          --console.ui.site-name="The Things Network Stack for LoRaWAN"
                                          --console.ui.sub-title="The official configuration platform for The Things Network"
                                        --console.ui.theme-color=""
                                              --console.ui.title="Console"
                                   --device-repository.directory=""
                                         --device-repository.url=""
                                                --events.backend="internal"
                                      --events.cloud.publish-url=""
                                    --events.cloud.subscribe-url=""
                                          --events.redis.address=""
                                         --events.redis.database="0"
                                        --events.redis.namespace=""
                                         --events.redis.password=""
                                     --frequency-plans.directory=""
                                           --frequency-plans.url="https://raw.githubusercontent.com/TheThingsNetwork/lorawan-frequency-plans/master"
                       --gcs.basic-station.allow-cups-uri-update="false"
              --gcs.basic-station.owner-for-unknown.account-type=""
                   --gcs.basic-station.owner-for-unknown.api-key=""
                        --gcs.basic-station.owner-for-unknown.id=""
                     --gcs.basic-station.require-explicit-enable="false"
                                              --gcs.require-auth="true"
                   --gcs.the-things-gateway.default.firmware-url="https://thethingsproducts.blob.core.windows.net/the-things-gateway/v1"
                    --gcs.the-things-gateway.default.mqtt-server=""
                 --gcs.the-things-gateway.default.update-channel="stable"
                           --grpc.allow-insecure-for-credentials="false"
                                                   --grpc.listen=":1884"
                                               --grpc.listen-tls=":8884"
                                             --gs.mqtt-v2.listen=""
                                         --gs.mqtt-v2.listen-tls=""
                                                --gs.mqtt.listen=":1882"
                                            --gs.mqtt.listen-tls=":8882"
                                --gs.require-registered-gateways="false"
                                      --gs.udp.addr-change-block="5m0s"
                                     --gs.udp.connection-expires="5m0s"
                                  --gs.udp.downlink-path-expires="30s"
                                              --gs.udp.listeners=":1700="
                                          --gs.udp.packet-buffer="50"
                                        --gs.udp.packet-handlers="10"
                                     --gs.udp.schedule-late-time="800ms"
                                         --http.cookie.block-key="38E31BCAD8CFC067ABC9F2988967E387E15DF9ADDA14E63F446ED955EEEA4637"
                                          --http.cookie.hash-key="40847E55ED0CB34B3D491DC557326BF875FCE34EE0C8F50194E1BB3488055FA96D5CC4F3CF6C30C5F4922D8CEB4F72A1FE61317E1A7BC88619617AD6CEA983B3"
                                            --http.health.enable="true"
                                          --http.health.password=""
                                                   --http.listen=":1885"
                                               --http.listen-tls=":8885"
                                           --http.metrics.enable="true"
                                         --http.metrics.password=""
                                             --http.pprof.enable="true"
                                           --http.pprof.password=""
                                             --http.static.mount="/assets"
                                       --http.static.search-path="public,/srv/ttn-lorawan/public"
                                  --is.auth-cache.membership-ttl="10m0s"
                                               --is.database-uri="postgresql://root@localhost:26257/ttn_lorawan_dev?sslmode=disable"
                                  --is.email.network.console-url="http://localhost:1885/console"
                          --is.email.network.identity-server-url="http://localhost:1885/oauth"
                                         --is.email.network.name="The Things Network Stack for LoRaWAN"
                                             --is.email.provider=""
                                       --is.email.sender-address=""
                                          --is.email.sender-name=""
                                     --is.email.sendgrid.api-key=""
                                     --is.email.sendgrid.sandbox="false"
                                         --is.email.smtp.address=""
                                     --is.email.smtp.connections="0"
                                        --is.email.smtp.password=""
                                        --is.email.smtp.username=""
                                                --is.oauth.mount=""
                                   --is.oauth.ui.assets-base-url="/assets"
                                     --is.oauth.ui.canonical-url="http://localhost:1885/oauth"
                                          --is.oauth.ui.css-file="oauth.css"
                                      --is.oauth.ui.descriptions=""
                                       --is.oauth.ui.icon-prefix="oauth-"
                                           --is.oauth.ui.js-file="oauth.js"
                                          --is.oauth.ui.language="en"
                                         --is.oauth.ui.site-name="The Things Network Stack for LoRaWAN"
                                         --is.oauth.ui.sub-title=""
                                       --is.oauth.ui.theme-color=""
                                             --is.oauth.ui.title=""
                                     --is.profile-picture.bucket="profile_pictures"
                                 --is.profile-picture.bucket-url="/assets/blob/profile_pictures"
                               --is.profile-picture.use-gravatar="true"
                  --is.user-registration.admin-approval.required="false"
         --is.user-registration.contact-info-validation.required="false"
                      --is.user-registration.invitation.required="false"
                     --is.user-registration.invitation.token-ttl="168h0m0s"
         --is.user-registration.password-requirements.min-digits="1"
         --is.user-registration.password-requirements.min-length="8"
        --is.user-registration.password-requirements.min-special="0"
      --is.user-registration.password-requirements.min-uppercase="1"
                                            --js.join-eui-prefix="0000000000000000/0"
                                              --key-vault.static=""
                                                     --log.level="info"
                                            --ns.cooldown-window="1s"
                                       --ns.deduplication-window="200ms"
                            --ns.default-mac-settings.adr-margin="15"
                       --ns.default-mac-settings.class-b-timeout="1m0s"
                       --ns.default-mac-settings.class-c-timeout="15s"
                     --ns.default-mac-settings.desired-rx1-delay="5"
              --ns.default-mac-settings.status-count-periodicity="20"
               --ns.default-mac-settings.status-time-periodicity="1h0m0s"
                                          --ns.dev-addr-prefixes=""
                            --ns.downlink-priorities.join-accept="highest"
                           --ns.downlink-priorities.mac-commands="highest"
               --ns.downlink-priorities.max-application-downlink="high"
                                                     --ns.net-id="000000"
                                                 --redis.address="localhost:6379"
                                                --redis.database="0"
                                               --redis.namespace="ttn,v3"
                                                --redis.password=""
                                                    --rights.ttl="2m0s"
                                                    --sentry.dsn=""
                                                  --tls.acme.dir=""
                                                --tls.acme.email=""
                                               --tls.acme.enable="false"
                                             --tls.acme.endpoint="https://acme-v01.api.letsencrypt.org/directory"
                                                --tls.acme.hosts=""
                                               --tls.certificate="cert.pem"
                                                       --tls.key="key.pem"
                                                   --tls.root-ca=""

#### What do you see now?

root@propc:/home/robert# curl http://localhost:1885/api/v3/as/applications/app3/webhooks/wh4/devices/dev4/down/push -X POST   -H 'Authorization: Bearer "NNSXS.GIPVAVIAO77RFBOWTDO43XIZ5SDTRPDX7QKOGCQ.EJJRVXWEX7LIFSPYMT4WBPPSCIMIJEXTRMFOSRNT47RPF6KTWCJQ'  --data '{"downlinks":[{"frm_payload":"vu8=","f_port":15,"priority":"NORMAL"}]}'

error:pkg/auth/rights:no_application_rights (no rights for application `app3`)

...

#### What do you want to see instead?

...

#### Environment

UBUNTU 16.04LTS/FF/iFemtoCell+ SPF/TELIT RE866
...

#### How do you propose to implement this?

...

#### Can you do this yourself and submit a Pull Request?

...

### LINKED ISSUE #850: Update multicast guide in getting started
#### Summary

Multicast device setup instructions should be updated now that #792 is merged.

#### Why do we need this?

To give better guidance

#### What is already there? What do you see now?

Outdated multicast device setup instructions

#### What is missing? What do you want to see?

Multicast device setup instructions should be updated.

#### Can you do this yourself and submit a Pull Request?

Will review

### PR DESCRIPTION
**Pull request by adriansmares**
#### Summary

Closes #845 
Closes #850 

#### Changes

- Add a note regarding the API key required for Webhooks downlink queue operations.
- Add multicast device creation instructions and point to them for class C multicast.

#### Notes for Reviewers

None.

### CODE REVIEW

File: doc/gettingstarted.md [hunk 1]
```diff
@@ -221,6 +221,22 @@ $ ttn-lw-cli end-devices create app1 dev2 \
...
+```bash
+$ ttn-lw-cli end-devices create app1 dev3 \
+  --frequency-plan-id EU_863_870 \
+  --lorawan-version 1.0.2 \
+  --lorawan-phy-version 1.0.2-b \
+  --abp \
+  --session.dev-addr 00E4304D \
+  --session.keys.app-s-key.key A0CAD5A30036DBE03096EB67CA975BAA \
+  --session.keys.nwk-s-key.key B7F3E161BC9D4388E6C788A0C547F255 \
+  --multicast
```
Comment (rvolosatovs) [inline doc/gettingstarted.md]: @htdvisser shouldn't this be `multicast=true` ?

File: doc/gettingstarted.md [hunk 2]
```diff
@@ -221,6 +221,22 @@ $ ttn-lw-cli end-devices create app1 dev2 \
...
 >Hint: You can also pass `--with-session` to have a session generated.
 
+It is also possible to create a multicast device (an ABP device which can not send uplinks and shares the security session with other devices) using the `--multicast` flag as follows:
+
+```bash
+$ ttn-lw-cli end-devices create app1 dev3 \
+  --frequency-plan-id EU_863_870 \
+  --lorawan-version 1.0.2 \
+  --lorawan-phy-version 1.0.2-b \
+  --abp \
```
Comment (rvolosatovs) [inline doc/gettingstarted.md]: This should not be necessary @htdvisser?

File: doc/gettingstarted.md (hunk 1 above)
Comment (adriansmares) [reply doc/gettingstarted.md]: Boolean flags can be set using just `--flag`, and they go to true by default.

```
adriansmares@A98BCD2222F1:~/go/src/go.thethings.network/lorawan-stack$ ttn-lw-cli end-devices create app1 dev2 \
>   --frequency-plan-id EU_863_870 \
>   --lorawan-version 1.0.2 \
>   --lorawan-phy-version 1.0.2-b \
>   --abp \
>   --session.dev-addr 00E4304D \
>   --session.keys.app-s-key.key A0CAD5A30036DBE03096EB67CA975BAA \
>   --session.keys.nwk-s-key.key B7F3E161BC9D4388E6C788A0C547F255 \
>   --multicast
Flag --session.keys.nwk-s-key.key has been deprecated, use the session.keys.f_nwk_s_int_key.key flag
 DEBUG Using access token (valid until 11:10AM)
 DEBUG ccResolverWrapper: sending update to cc: {[{localhost:1884 0  <nil>}] }
 DEBUG pickfirstBalancer: HandleSubConnStateChange: 0xc0005ddc00, CONNECTING
 DEBUG pickfirstBalancer: HandleSubConnStateChange: 0xc0005ddc00, READY
 DEBUG Finished unary call                      duration=44.494674ms grpc_method=Create grpc_service=ttn.lorawan.v3.EndDeviceRegistry namespace=grpc
 DEBUG Set end device on Network Server         paths=[frequency_plan_id lorawan_phy_version lorawan_version multicast session.dev_addr session.keys.f_nwk_s_int_key.key supports_join]
 DEBUG Finished unary call                      duration=455.538557ms grpc_method=Set grpc_service=ttn.lorawan.v3.NsEndDeviceRegistry namespace=grpc
 DEBUG Set end device on Application Server     paths=[session.dev_addr session.keys.app_s_key.key]
 DEBUG Finished unary call                      duration=48.61551ms grpc_method=Set grpc_service=ttn.lorawan.v3.AsEndDeviceRegistry namespace=grpc
{
  "ids": {
    "device_id": "dev2",
    "application_ids": {
      "application_id": "app1"
    }
  },
  "created_at": "2019-07-01T08:11:48.868Z",
  "updated_at": "2019-07-01T08:11:49.381217500Z",
  "attributes": {
  },
  "network_server_address": "localhost",
  "application_server_address": "localhost",
  "lorawan_version": "1.0.2",
  "lorawan_phy_version": "1.0.2-b",
  "frequency_plan_id": "EU_863_870",
  "session": {
    "dev_addr": "00E4304D",
    "keys": {
      "f_nwk_s_int_key": {
        "key": "B7F3E161BC9D4388E6C788A0C547F255"
      },
      "app_s_key": {
        "key": "A0CAD5A30036DBE03096EB67CA975BAA"
      }
    },
    "started_at": "0001-01-01T00:00:00Z"
  },
  "multicast": true
}
```

File: doc/gettingstarted.md (hunk 2 above)
Comment (adriansmares) [reply doc/gettingstarted.md]: If `--abp` is not specified, `supports_join` is set to true. See [this](https://github.com/TheThingsNetwork/lorawan-stack/blob/861fe5eaec412b8da0ff47cdb431b41e49a30b50/cmd/ttn-lw-cli/commands/end_devices.go#L328-L367).
We could also check if its either `--abp` or `--multicast`.

Comment (rvolosatovs) [reply doc/gettingstarted.md]: Can you file an issue about it?

Comment (adriansmares) [reply doc/gettingstarted.md]: #889

**Comment by adriansmares** [comment]
Waiting on @johanstokking for Webhooks wording (although it believe it is fine).

**Pull request merged**